driver/src/price_finding/price_finder_interface.rs

```diff
@@ -23,37 +20,35 @@ impl Default for Fee {
 }
 
 #[derive(Clone, Debug, Copy, PartialEq)]
-pub enum SolverType {
+pub enum SolverConfig {
     NaiveSolver,
-    StandardSolver,
-    FallbackSolver,
+    StandardSolver(u32), //params for --solverTimeLimit
+    FallbackSolver(u32),
 }
 
-impl FromStr for SolverType {
-    type Err = Infallible;
-
-    fn from_str(solver_type_str: &str) -> Result<Self, Self::Err> {
+impl SolverConfig {
+    pub fn new(solver_type_str: &str, solver_time_limit: u32) -> Self {
         match solver_type_str.to_lowercase().as_str() {
-            "standard-solver" => Ok(SolverType::StandardSolver),
-            "fallback-solver" => Ok(SolverType::FallbackSolver),
-            // the naive solver is the standard solver.
-            _ => Ok(SolverType::NaiveSolver),
+            "standard-solver" => SolverConfig::StandardSolver(solver_time_limit),
+            "fallback-solver" => SolverConfig::FallbackSolver(solver_time_limit),
+            "naive-solver" => SolverConfig::NaiveSolver,
+            _ => panic!("solver type does not exist"),
```

Comment:
Nit: It is cleaner to continue to have this function return `Result`. Then we can panic at the point where that result is needed by unwrapping. To do this we would return `anyhow::Result<Self>` in this function and then `Err(anyhow!("solver type does not exit"))` instead of panic.